### PR TITLE
[W5.4][F11-A4] Han Yaodong

### PR DESCRIPTION
**Pull request by HanYaodong**
Add a new sort command to the address, which will show a list of people in alphabetical order. Add I/O test and JUnit test to the sort command.

### CODE REVIEW

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 1]
```diff
@@ -81,6 +82,21 @@ public UniquePersonList(UniquePersonList source) {
...
     }
 
+    /**
+     *  Returns an unmodifiable java List view with elements sorted in alphabetical order.
+     *  All the elements are cast as immutable {@link ReadOnlyPerson}s.
+     *  Any changes to the internal list/elements are immediately visible in the returned list.
+     */
+    public List<ReadOnlyPerson> getSortedListView() {
+        List<Person> sortedList = new ArrayList<>(internalList);
+        sortedList.sort(new Comparator<Person>() {
```
Comment (syy94) [inline src/seedu/addressbook/data/person/UniquePersonList.java]: Try the lambda expression version?

Some help here:
https://dzone.com/articles/using-lambda-expression-sort

Comment (HanYaodong) [reply src/seedu/addressbook/data/person/UniquePersonList.java]: Noted and changed!

**Comment by yewshengkai** [comment]
As mentioned before, maybe we can consider the following:
+Sort, re-generate and save the list in the sorted format (Can consider SYY94 command to select Y/N for saving).
+Sort by other method other than alphabetical order (e.g. Age, birth-date, etc - If we had these extra data).

**Comment by damithc** [comment]
@cmkumar87 reminder to review this PR

File: test/java/seedu/addressbook/commands/SortCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+public class SortCommandTest {
+
+    private AddressBook addressBook;
+    private List<ReadOnlyPerson> listInSortedOrder;
+    private List<ReadOnlyPerson> emptyList;
+
+    @Before
+    public void setUp() throws Exception {
+
```
Comment (cmkumar87) [inline test/java/seedu/addressbook/commands/SortCommandTest.java]: Blank line needed here? Check coding guidelines?

File: test/java/seedu/addressbook/commands/SortCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        emptyList = TestUtil.createList();
+
+    }
+
+    @Test
+    public void execute() throws Exception {
+        assertSortInRightOrder(this.addressBook, this.listInSortedOrder);
+    }
+
+    private void assertSortInRightOrder(AddressBook addressBook, List<ReadOnlyPerson> listInSortedOrder) {
```
Comment (cmkumar87) [inline test/java/seedu/addressbook/commands/SortCommandTest.java]: dont understand what is the diff between this method and the helper it calls!! looks like all you need is just 1 method; the method that is current a helper

File: doc/UserGuide.md [hunk 4]
```diff
@@ -53,6 +53,10 @@ Examples:
 Shows a list of all persons in the address book.<br>
 Format: `list`
```
Comment (cmkumar87) [inline doc/UserGuide.md]: good! continue to update user guides and help docs during your project development phases too.

**Pull request closed without merging**